ProcessMaker/processmaker
# FOUR-9028 | Add a Custom Icon

# Issue
Ticket: [FOUR-9028](https://processmaker.atlassian.net/browse/FOUR-9028)

As PM Block Author, I want to select a custom icon for my Block, so that it shows in the modeler’s PM Block task when it is added to a process. I should be able to choose from the default selection of available icons, or upload my own. 

The PM Block Configuration tab should have a dedicated field called “Icon” in order to allow for this selection.

The Icon should be seen in the PM Block browsing tab within the Modeler’s right rail, to the left of the PM Block name/title.

# How to Test
1. Go to branch `feature/FOUR-9028` in core.
2. Go to branch `feature/FOUR-9028` in `package-pm-blocks`.
3. Go to branch `feature/FOUR-9028` in `modeler`.
4. In the PM Blocks configuration, you should be able to save a custom icon to a PM Block.
5. When creating a PM Block, you should be able to save a custom icon to the PM Block.
6. Within the Modeler, the PM Block custom icon should display in the PM Block listing in the right rail.

# Related Tickets & Packages
- [FOUR-7365](https://processmaker.atlassian.net/browse/FOUR-7365) | PM Blocks

# Code Review Checklist


[FOUR-9028]: https://processmaker.atlassian.net/browse/FOUR-9028?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-7365]: https://processmaker.atlassian.net/browse/FOUR-7365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
